include/up-cpp/utils/Expected.h

```diff
@@ -27,82 +28,137 @@ static_assert(!__has_cpp_attribute(__cpp_lib_expected),
 ///          https://en.cppreference.com/w/cpp/utility/expected
 ///          No further documentation is provided in this file.
 /// @{
-template <typename E>
-struct BadExpectedAccess;
+struct BadExpectedAccess : public std::runtime_error {
+	BadExpectedAccess() : std::runtime_error("generic BadExpectedAccess") {}
 
-template <>
-struct BadExpectedAccess<void> : public std::exception {};
+	template <typename... Args>
+	BadExpectedAccess(Args&&... args)
+	    : std::runtime_error(std::forward<Args>(args)...) {}
+};
 
+/// @brief Required tagging type for cases where expected and unexpected type
+/// are identical.
 template <typename E>
-struct BadExpectedAccess : public BadExpectedAccess<void> {};
+class Unexpected {
+public:
```

Comment:
Redundant access specifier
```suggestion
struct Unexpected {
```